federationapi/routing/state.go

```diff
@@ -102,6 +102,10 @@ func getState(
 		return nil, resErr
 	}
 
+	if event.RoomID() != roomID {
```

Comment:
this is my first review comment, looks ok but I would expect to see handler tests

Reply:
Thanks for the message; sorry it's been some delay I came back to this issue. But I'm not very sure what "handler tests" refer to. :/ Would you mind giving me some hints?

Reply:
Possibly unit tests?